enterprise/views.py

```diff
@@ -349,10 +359,14 @@ def post_account_consent(self, request, consent_provided):
         """
         self.lift_quarantine(request)
 
-        # Load the linked EnterpriseCustomer for this request. Return a 404 if no such EnterpriseCustomer exists
+        # Load the linked EnterpriseCustomer for this request.
```

Comment:
@mattdrayer I don't see a nice way to refactor this. For example, we can't combine the logic for obtaining the enterprise customer for the current request with the redirect logic:

```python
customer = self.get_customer_or_redirect(request)
```

... because we'd get different types of values for `customer` depending on the state of the workflow.

If we only move the redirect logic to a separate method:

```python
customer = get_enterprise_customer_for_request(request)
self.redirect_if_no_customer(customer)
```

... I feel like we're not gaining much in terms of DRYness and it becomes less obvious that the method might short-circuit at this step.